Microsoft.Exchange.WebServices.Tests/WithDnsClient.cs

```diff
@@ -69,6 +69,29 @@ public void SrvQuery_Should_Execute_Srv_Query_For_Specified_Domain()
         dnsQueryClientMock.VerifyAll(); // Verifying strict mock.
     }
 
+    /// <remarks>
+    /// Could happen on DNS query timeout.
+    /// </remarks>
+    [Test]
+    public void SrvQuery_Should_Throw_DnsException_On_DnsResponseException()
+    {
+        // Arrange
+        var dnsQueryClientMock = fixture.Freeze<Mock<IDnsQuery>>();
+        var srvDomain = "any.domain";
+        var errorMessage = "some error message";
+
+        dnsQueryClientMock
+            .Setup(dnsQueryClient => dnsQueryClient.Query(srvDomain, QueryType.SRV, QueryClass.IN))
+            .Throws(new DnsResponseException(errorMessage));
+
+        // Act
+        void InvokeSrvQuery() => dnsClient.SrvQuery(domain: "any.domain");
+
+        // Assert
+        var dnsException = Assert.Throws<DnsException>(InvokeSrvQuery);
+        Assert.AreEqual(expected: errorMessage, actual: dnsException!.Message);
```

Comment:
Shouldn't we check inner exception as well?

Reply:
sure